presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -338,6 +333,30 @@ public synchronized void process(Iterable<QueryExecution> runningQueries, Suppli
         updateNodes(assignmentsRequest);
     }
 
+    public synchronized void checkForLeaks(Supplier<List<BasicQueryInfo>> allQueryInfoSupplier)
+    {
+        if (!enabled) {
+            return;
+        }
+
+        Map<QueryId, Optional<BasicQueryInfo>> allRunningQueries;
+        if (memoryManagerService.isPresent()) {
+            // We are in the multi-coordinator codepath, and thus care about the globally running queries
+            allRunningQueries = getClusterInfo(GENERAL_POOL)
+                            .getRunningQueries()
+                            .orElse(Collections.emptyList()).stream().collect(
+                                Collectors.toMap(Function.identity(), t -> Optional.empty()));
```

Comment:
```suggestion
            allRunningQueries = getClusterInfo(GENERAL_POOL)
                            .getRunningQueries()
                            .orElse(ImmutableList.of())
                            .stream()
                            .collect(toImmutableMap(identity(), t -> Optional.empty()));
```

Let's static import the utility methods, and converge on immutable list for the empty list.